Simplified the file filter predication code generation.

stuartskelton opened the pull request:
Ok, I may be a little guilty of golfing a little here, but this removes the brain ache of trying to keep track of what has been set or not.


The only reason I could get away with this is because all the checks were ANDs.

stevan commented:
Looks mostly good to me, I am adding a few small comments in the code for discussion.

stevan commented on lib/App/Critique/Command/collect.pm:
```diff
@@ -179,98 +179,25 @@ sub generate_file_predicate {
+    # the files, the code refs assume the params will be $path,$rel.
+    my @filters = (sub { return 1 });
+    push @filters, sub { return $_[1] !~ /$filter/} if $filter;
+    push @filters, sub { return $_[1] =~ /$match/ } if $match ;
```
I think the order of `filter` and `match` here is reversed, if you look at the old code, `match` always precedes `filter'.

stuartskelton replied:
you are very correct

stevan commented on lib/App/Critique/Command/collect.pm:
```diff
@@ -179,98 +179,25 @@ sub generate_file_predicate {
-    # ------------------------------#
-    #    0   |   0   |      1       # collect with no violations
-    # ------------------------------#
-
```
I would like to retain this table in the new version, just as documentation.

stevan commented:
Okay, comments added, one question and one request.

stuartskelton commented:
I have corrected the filters.

and:

I have also added back the table, however with a little change in the ordering, so the stages descend in their filters binary value, I think it makes it a little more readable, but I am happy to change it back if you disagree.

stuartskelton commented:
Thinking about it too, I should probably switch Cols 1 and 2. too as that would fit with the logic.

stevan commented:
Yeah, switching cols 1 and 2 would be good, I agree, after that, this patch is good, thanks!

stuartskelton commented:
I have amended the last commit.
